src/TheThingsNetwork.cpp

```diff
@@ -280,29 +289,39 @@ bool TheThingsNetwork::personalize() {
 }
 
 bool TheThingsNetwork::provision(const char *appEui, const char *appKey) {
-  sendMacSet(MAC_SET_APPEUI, appEui);
-  sendMacSet(MAC_SET_APPKEY, appKey);
-  debugPrint(SENDING);
-  sendCommand(MAC_TABLE, MAC_PREFIX, true);
-  sendCommand(MAC_TABLE, MAC_SAVE, false);
-  modemStream->write(SEND_MSG);
-  debugPrintLn();
-  return true;
+  if (bufLength(appEui) == 16 && bufLength(appKey) == 32) {
+    sendMacSet(MAC_SET_APPEUI, appEui);
+    sendMacSet(MAC_SET_APPKEY, appKey);
+    debugPrint(SENDING);
+    sendCommand(MAC_TABLE, MAC_PREFIX, true);
+    sendCommand(MAC_TABLE, MAC_SAVE, false);
+    modemStream->write(SEND_MSG);
+    debugPrintLn();
+    return true;
+  }
```

Comment:
@Nicolasdejean could you move `errMessage(JOIN_INCORRECT_KEYS)` to `provision()`? then I'll test and we can merge.